src/pipecat/audio/turn/base_turn_analyzer.py

```diff
@@ -45,11 +51,23 @@ def set_sample_rate(self, sample_rate: int):
         """
         self._sample_rate = self._init_sample_rate or sample_rate
 
+    @property
+    def last_prediction_result(self) -> Optional[Dict[str, Any]]:
+        """Get and clear the last prediction result.
+
+        This method should be implemented by subclasses to provide access
+        to the most recent prediction result. The default implementation
+        returns None.
+
+        Returns:
+            Optional[Dict[str, Any]]: The last prediction result, if any.
+        """
+        return None
```

Comment:
Do we nerd this function? Who is supposef to use it? Also, maybe it shouldn’t clear the value so you can always access to it.